### PR TITLE
Add a makeshift state machine

### PR DESCRIPTION
**Pull request by mbklein**
This PR sprinkles some sugar on top of the actor/job stack to make it possible to interrogate "doneness" better than we could before (though still not perfectly).

- Add a makeshift state machine that automatically transitions when jobs are enqueued and performed
- Add `Image#processed?` and `FileSet#processed?` to check if a specific set of states have transitioned to `performed`

partially fixes https://github.com/nulib/next-generation-repository/issues/758

### CODE REVIEW

File: db/migrate/20181023142014_create_crappy_state_machines.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateCrappyStateMachines < ActiveRecord::Migration[5.1]
+  def change
+    create_table :crappy_state_machine do |t|
```
Comment (kdid) [inline db/migrate/20181023142014_create_crappy_state_machines.rb]:
```suggestion
    create_table :crappy_state_machines do |t|
```

File: app/models/file_set.rb [hunk 2]
```diff
@@ -7,4 +7,9 @@ class FileSet < ActiveFedora::Base
   def to_common_index
     CommonIndexService.index(self)
   end
+
+  def processed?
+    file_set_jobs = ['CharacterizeJob', 'CreateDerivativesJob', 'CreatePyramidTiffJob', 'IngestJob']
```
Comment (kdid) [inline app/models/file_set.rb]: If we use the state machine, I think we are going to care about the `VisibilityCopyJob` and
`InheritPermissionsJob` as well. I know that users are going to use the UI to adjust the visibility and that will affect what appears in the index, manifest and front end.

**Comment by davidschober** [comment]
I believe we're moving past this

**Comment by kdid** [comment]
closing. We are pivoting. For now

**Pull request closed without merging**